test/helpers/common.js

```diff
@@ -870,37 +870,77 @@ export function swapDisplayedColumns(container, from, to) {
   $from.simulate('mouseup');
 }
 
+/**
+ * Creates touch event and dispatch it for handled element.
+ *
+ * @param {number} x The page x coordinates.
+ * @param {number} y The page y coordinates.
+ * @param {HTMLElement} element An element for which event will be triggered.
+ * @param {string} eventType Type of touch event, ie. 'touchstart', 'touchmove', 'touchend'.
+ * @returns {boolean} The return value is `false` if event is cancelable and at least one of the event handlers which
+ * received event called `preventDefault()`. Otherwise it returns `true`.
+ */
+function sendTouchEvent(x, y, element, eventType) {
+  const touchObj = new Touch({
+    identifier: Date.now(),
+    target: element,
+    clientX: x,
+    clientY: y,
+    radiusX: 2.5,
+    radiusY: 2.5,
+  });
+
+  const touchEvent = new TouchEvent(eventType, {
+    cancelable: true,
+    bubbles: true,
+    touches: eventType === 'touchend' ? [] : [touchObj],
+    targetTouches: eventType === 'touchend' ? [] : [touchObj],
+    changedTouches: [touchObj],
+    shiftKey: false,
+  });
+
+  return element.dispatchEvent(touchEvent);
+}
+
 /**
  * @param {string} type A name/type of the event.
  * @param {HTMLElement} target The target element from the event was triggered.
  * @param {number} pageX The page x coordinates.
  * @param {number} pageY The page y coordinates.
+ * @returns {boolean} The return value is `false` if event is cancelable and at least one of the event handlers which
+ * received event called `preventDefault()`. Otherwise it returns `true`.
  */
 export function triggerTouchEvent(type, target, pageX, pageY) {
-  const e = document.createEvent('TouchEvent');
-
   const targetCoords = target.getBoundingClientRect();
-  const targetPageX = pageX || parseInt(targetCoords.left + 3, 10);
-  const targetPageY = pageY || parseInt(targetCoords.top + 3, 10);
-  let touches;
-  let targetTouches;
-  let changedTouches;
-
-  const touch = document.createTouch(window, target, 0, targetPageX, targetPageY, targetPageX, targetPageY);
-
-  if (type === 'touchend') {
-    touches = document.createTouchList();
-    targetTouches = document.createTouchList();
-    changedTouches = document.createTouchList(touch);
-  } else {
-    touches = document.createTouchList(touch);
-    targetTouches = document.createTouchList(touch);
-    changedTouches = document.createTouchList(touch);
-  }
+  const targetPageX = pageX || parseInt(targetCoords.left, 10) + 3;
+  const targetPageY = pageY || parseInt(targetCoords.top, 10) + 3;
```

Comment:
```suggestion
  const targetPageX = pageX ?? parseInt(targetCoords.left, 10) + 3;
  const targetPageY = pageY ?? parseInt(targetCoords.top, 10) + 3;
```
Otherwise, there is no possibility to set page coordinates as `0` :)